train.py

```diff
@@ -623,7 +660,11 @@ def main(args):
         timer("fwd-bwd").stop()
 
         # update parameters, and returns (success_update, grad_norm)
-        trainer_result = trainer.step()
+        need_save_ckpt = (
```

Comment:
Internlm is [here](https://github.com/InternLM/InternLM/blob/4832671abe4abf01c3d2339ed4e77c961fb3770e/internlm/utils/model_checkpoint.py#L356C19-L356C19) to determine whether to store ckpt, including snapshots and signal handlers. When these conditions are met, we all need to synchronize forward overlap.

Reply:
done